tests/unit/integration_test.py

```diff
@@ -124,11 +126,54 @@ def test_rfi_get_size() -> None:
         }
     }
     error = RecordFluxError()
-    integration.add_integration_file(Path("p.rfi"), session_object, error)
+    # pylint: disable = protected-access
+    integration._add_integration_object(Path("p.rfi"), session_object, error)
     error.propagate()
     assert integration.get_size(ID("P::S"), ID("x"), ID("S")) == 1024
     assert integration.get_size(ID("P::S"), ID("x"), ID("S")) == 1024
     assert integration.get_size(ID("P::S"), ID("x"), None) == 1024
     assert integration.get_size(ID("P::S2"), ID("x"), None) == 4096
     assert integration.get_size(ID("P::S"), ID("y"), None) == 2048
     assert integration.get_size(ID("P::S"), ID("y"), ID("S")) == 8192
+
+
+@pytest.mark.parametrize(
+    "content, error_msg, line, column",
+    [
+        ('"', ["while scanning a quoted scalar", "unexpected end of stream"], 1, 2),
+        ("Session: 1, Session : 1", ["mapping values are not allowed here"], 1, 21),
+        (
+            "Session: 1\nSession : 1",
+            ["while constructing a mapping", 'found duplicate key "Session" with value "1"'],
+            2,
+            1,
+        ),
+    ],
+)
+def test_load_integration_file(
+    tmp_path: Path, content: str, error_msg: Sequence[str], line: int, column: int
+) -> None:
+    test_rfi = tmp_path / "test.rfi"
+    test_rfi.write_text(content)
+    integration = Integration()
+    error = RecordFluxError()
+    regex = fr"^{test_rfi}:{line}:{column}: parser: error: "
+    for elt in error_msg:
+        regex += elt
+        regex += fr'.*in "{test_rfi}", line [0-9]+, column [0-9]+.*'
+    regex += "$"
+    compiled_regex = re.compile(regex, re.DOTALL)
+    with pytest.raises(RecordFluxError, match=compiled_regex):
+        integration.load_integration_file(test_rfi, error)
+        error.propagate()
+
+
+def test_load_integration_path(tmp_path: Path) -> None:
+    subfolder = tmp_path / "sub"
+    subfolder.mkdir()
+    test_rfi = subfolder / "test.rfi"
+    test_rfi.write_text("{ Session: { Session : { Buffer_Size : {} }}}")
+    integration = Integration(integration_files_dir=subfolder)
+    error = RecordFluxError()
+    integration.load_integration_file(tmp_path / "test.rflx", error)
+    error.propagate()
```

Comment:
What functionality is tested here? The test would still be successful, even if `load_integration_file` would ignore `sub/test.rfi`.

Reply:
Indeed I was trying to avoid having to match an error message again, but this makes the test weaker.  Fixed.